WISE-Community/WISE-Client
# feat(Authoring): Add JSON authoring view for when project is broken

## Changes

- Added recovery view to allow editing the project JSON without loading the project
- The recovery view also tries to detect and display potential problems

## Test

1. Go to the recovery authoring view for a project.
```http://localhost:81/teacher/edit/unit/<unit id>/recovery```
2. Modify the JSON so that it becomes invalid, the upper right message should say "JSON Invalid" and the "Save" button should be disabled.
3. Modify the JSON so that it becomes valid, the upper right message should say "JSON Valid" and the "Save" button should be enabled.
4. Modify the project so that a node has a null transition and make sure the potential problem is detected.
```
        {
            "id": "node1",
            "transitionLogic": {
                "transitions": [
                    {
                        "to": null
                    }
                ]
            }
        }
```
5. Modify the project so that a group references a node id that does not exist and make sure the potential problem is detected.
```
        {
            "id": "group1",
            "ids": [
                "node1",
                "node9999"
            ]
        }
```
6. Modify the project so that a group references a node id multiple times and make sure the potential problem is detected.
```
        {
            "id": "group1",
            "ids": [
                "node1",
                "node1"
            ]
        }
```
7. Make sure the "Go to Authoring View" button works if the project has no problems.

Closes #922
